artemis-server/src/main/java/org/apache/activemq/artemis/core/protocol/core/ServerSessionPacketHandler.java

```diff
@@ -635,6 +648,27 @@ private void slowPacketHandler(final Packet packet) {
                   }
                   break;
                }
+               case CREATE_PRODUCER: {
+                  CreateProducerMessage message = (CreateProducerMessage) packet;
+                  // this is used to create/destroy the producer so needs to be unique
+                  String senderName = null;
+                  if (!producers.containsKey(message.getId())) {
+                     senderName = PRODUCER_ID_PREFIX + UUIDGenerator.getInstance().generateUUID();
+                     producers.put(message.getId(), senderName);
+                  }
+                  session.addProducer(senderName, ActiveMQClient.DEFAULT_CORE_PROTOCOL, message.getAddress() != null ? message.getAddress().toString() : null);
```

Comment:
This would still call addProducer even supposing an id was reused, but would call it with null senderName. Should probably not call it, or log or throw?